site/docs/glossary.md

```diff
@@ -0,0 +1,545 @@
+---
+layout: documentation
+title: Bazel Glossary
+---
+
+# Bazel Glossary
```

Comment:
Some of these have extensive "dive in deeper" docs elsewhere (e.g. cquery or platforms). What should be the process for that?

* Have a standardized **See also:** section at the bottom of each entry?
* Arbitrary inline text / links?
* Don't link out at all?

Reply:
We should link out, and I like the *See also:* approach. I'll add link outs in a follow up PR to keep this iteration scoped down.

Reply:
SGTM. Happy to help with the content of them if you need.